src/nick.c

```diff
@@ -34,7 +34,7 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 User *find_account(const Nick *nptr)
 {
-    return *nptr->svid ? find_user(nptr->svid) : find_user(nptr->nick);
+    return *nptr->svid != "0" ? find_user(nptr->svid) : find_user(nptr->nick);
```

Comment:
Is this actually compiling without warning? *nptr->svid is a "char" and "0" is a "char *". What you want here is:
... *nptr->svid != '0' (notice the single quotes). For the sake of thoroughness, we might just want to do ... (*nptr->svid && strcmp(nptr->svid, "0")) so that we are resilient against (i) nptr->svid still zeroed from initial allocation and (ii) valid svid starting with a 0 (it shouldn't exist though, but who knows).